src/main/kotlin/no/nav/familie/ef/sak/behandlingsflyt/steg/BeregnYtelseSteg.kt

```diff
@@ -93,35 +100,67 @@ class BeregnYtelseSteg(private val tilkjentYtelseService: TilkjentYtelseService,
                     }
 
     private fun andelerForInnvilgetRevurdering(behandling: Behandling,
-                                               beløpsperioder: List<AndelTilkjentYtelse>): List<AndelTilkjentYtelse> {
+                                               beløpsperioder: List<AndelTilkjentYtelse>,
+                                               opphørsperioder: List<Periode>): List<AndelTilkjentYtelse> {
         return behandling.forrigeBehandlingId?.let {
             val forrigeTilkjenteYtelse = hentForrigeTilkjenteYtelse(behandling)
-            return slåSammenAndelerSomSkalVidereføres(beløpsperioder, forrigeTilkjenteYtelse)
+            return slåSammenAndelerSomSkalVidereføres(beløpsperioder, forrigeTilkjenteYtelse, opphørsperioder)
         } ?: beløpsperioder
     }
 
     fun slåSammenAndelerSomSkalVidereføres(beløpsperioder: List<AndelTilkjentYtelse>,
-                                           forrigeTilkjentYtelse: TilkjentYtelse): List<AndelTilkjentYtelse> {
+                                           forrigeTilkjentYtelse: TilkjentYtelse,
+                                           opphørsperioder: List<Periode>): List<AndelTilkjentYtelse> {
         val fom = beløpsperioder.first().stønadFom
-        return forrigeTilkjentYtelse.taMedAndelerFremTilDato(fom) + beløpsperioder
+        val nyePerioderUtenOpphør = forrigeTilkjentYtelse.taMedAndelerFremTilDato(fom) + beløpsperioder
+        return vurderPeriodeForOpphør(nyePerioderUtenOpphør, opphørsperioder)
+
+    }
+
+    fun vurderPeriodeForOpphør(andelTilkjentYtelser: List<AndelTilkjentYtelse>,
+                               opphørsperioder: List<Periode>): List<AndelTilkjentYtelse> {
+        return andelTilkjentYtelser.map {
+
+            val tilkjentPeriode = it.periode
+            if (opphørsperioder.none { periode -> periode.overlapper(tilkjentPeriode) }) {
+                listOf(it)
+            } else if (opphørsperioder.any { periode -> periode.inneholder(tilkjentPeriode) }) {
+                listOf()
+            } else {
+                val overlappendeOpphør = opphørsperioder.first { periode -> periode.overlapper(tilkjentPeriode) }
+
+                if (overlappendeOpphør.overlapperIStartenAv(tilkjentPeriode)) {
+                    vurderPeriodeForOpphør(listOf(it.copy(stønadFom = overlappendeOpphør.tildato.plusDays(1))), opphørsperioder)
+                } else if (overlappendeOpphør.overlapperISluttenAv(tilkjentPeriode)) {
+                    vurderPeriodeForOpphør(listOf(it.copy(stønadTom = overlappendeOpphør.fradato.minusDays(1))), opphørsperioder)
+                } else { // periodew blir delt i to av opphold.
+                    vurderPeriodeForOpphør(listOf(it.copy(stønadTom = overlappendeOpphør.fradato.minusDays(1)),
+                                                  it.copy(stønadFom = overlappendeOpphør.tildato.plusDays(1))), opphørsperioder)
```

Comment:
Sliter litt med rekursjonen her, men prøver vi å bumpe én dag frem/tilbake for å finne `breaking` tidspunkt?

Reply:
Skulle gjerne sett noen tester på denne. 

Eks;
```
Forrige:
|----A----|-----B----|----C---|

Ny:
.........|---O---|---B'---|-O-|-C'-|

Etter sammenslåing
|---A-|             |---B'---|      |-C'-|
```

Reply:
Jeg har laget den litt mer avansert enn nødvendig, fordi jeg misforstod litt. Nå skal koden takle at du sender inn ny stønadsperiode, med unntak. Vet at det ikke var helt det du tenkte, men det kan være en funksjon som er verdt noe. Slik denn er nå vil den takle at du sender inn ny andel 1.4 - 31.12 og opphold 1.5-31.5 og 1.8-31.8. dete vil resultere i nye andeler 1.4-30.40 og 1.6-31.7 og 1.9-31.12. Rekursjonen går videre og vurderer de nye periodene som kommer etter oppslitting mot oppholdene som skal legges inn. Tanken var da at saksbehandler kunnne legge inn enn lengre periode med ytelse og opphold som overlapper for unntak.

Reply:
Aha, I see! Fin forklaring - kan sikkert si det til Lars også. Det gir litt mer valgfrihet på hvordan han ønsker GUI-et her :)